File: Python Course/Software Development Module - in Python/Week6/AddLoopApp.py
"""
AddLoopApp.py
Billy Cussen
28/09/2020
"""

import gc
import AddLoop
from AddLoop import myLoop

repeat = "y"

while repeat == "y":

    inputCheck = True

    while inputCheck:
        try:
            number = int(input("Please enter a Number: "))
        except:
            print("You need to enter a number!")
        else:
            if(number <= 0):
                print("You need to enter a number greater than 0")
            else:
                inputCheck = False

    myLoopApp = myLoop(number)
    myLoopApp.executeLoop()
    myLoopApp = None
    gc.collect()

    repeat = ""

    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter Y or N to repeat: "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print("You need to enter \"Y\" or \"N\"!")
        
print("Thanks for using my App")

File: Java/Software Development Module - Java/Week6/AddLoopApp.java
/*
AddLoopApp.java
Billy Cussen
14/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class AddLoopApp {
    
    public static void main(String [] args){

        //Variables
        int input = 0;
        int output;
        boolean inputCheckAddLoop;

        //Object
        Scanner keyboard;
        AddLoop myAddLoop;

        do{
            inputCheckAddLoop = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter a number: ");
                input = keyboard.nextInt();
            }catch(InputMismatchException e){
                inputCheckAddLoop = false;
                System.out.println("You need to enter an integer");
            }finally{
                keyboard = null;
            }

        }while(!inputCheckAddLoop);

        myAddLoop = new AddLoop(input);
        output = myAddLoop.getOutput();
        myAddLoop = null;

        System.out.println("The result is: "+output);

    }

}